tmt/steps/provision/__init__.py

```diff
@@ -479,6 +479,19 @@ def stop(self):
         """
         self.debug(f"Doing nothing to stop guest '{self.guest}'.")
 
+    def reboot(hard=True):
```

Comment:
hard=True is not nice by default however here:
https://github.com/psss/tmt/issues/281#issuecomment-847820121
it looked like it's preferred default.

Reply:
That was just an initial brainstorming. When thinking about it a bit more, what about something like this?

    guest.reboot(how="soft")
    guest.reboot(how="hard")
    guest.reboot(how="best")

The last option (would probably need a better name) would be the default method and could be used to try a soft reboot and if it does not work or is not supported by the provision method, hard reboot would be used. @lukaszachy, @FrNecas, @thrix, @RHEmployee, what do you think?

@RHEmployee, do you plan to include at least an initial implementation of the reboot functionality as part of this pull request?

Reply:
Hi.

I already started implementation in minute plugin in our internal repo.

If "best" option makes sense then why not. But I'm not sure how to (in case of minute plugin) guarantee that reboot has been successful. But maybe just sending request and receive answer would be ok... but still not enough because OS can be stuck on something like blocking I/O etc...

Reply:
About initial implementation:
what do you think is needed to add in reboot() method?
I need to solve how to pass information about reboot to provision step of minute plugin first but I'm not sure what else is needed. Implementation in minute plugin aims to just passing data about reboots (still fighting with instance name - maybe another cli opt will help) and reboot itself.